.migration/docs/setup-guide.md

```diff
@@ -0,0 +1,204 @@
+# S2DM Migration Workflow Setup Guide
+
+This guide explains how to set up the automated S2DM migration workflow in your repository. The workflow automatically generates artifacts (GraphQL schemas, JSON schemas, registry files, and SKOS RDF) when changes are made to your specification files.
+
+## Prerequisites
+
+- A repository with S2DM specification files in a `spec/` directory
+- A `units.yaml` file at the repository root
+- GitHub Actions enabled in your repository
+
+## Setup Steps
+
+### 1. Copy Files from S2DM Repository
+
+Copy the required files from the [COVESA/s2dm](https://github.com/COVESA/s2dm) repository to your repository:
+
+#### Option A: Using the setup script (recommended)
+
+Run the automated setup script that handles everything for you:
+
+```bash
+curl -fsSL https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/setup-repository.sh | bash
+
+curl -fsSL https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/setup-repository.sh | bash -s -- --directory /path/to/your/project
+```
+
+This script will:
+
+- Initialize a git repository if needed
+- Create the `spec/` directory
+- Download the required workflow and configuration files
+- Handle existing files with confirmation prompts
+
+#### Option B: Download files directly from GitHub
+
+1. Download the workflow file:
+   - Go to https://github.com/COVESA/s2dm/blob/main/.migration/workflows/migrate.yml
+   - Click "Raw" and save the file to `.github/workflows/migrate.yml` in your repository
+
+2. Download the version bump configuration:
+   - Go to https://github.com/COVESA/s2dm/blob/main/.migration/.bumpversion.toml
+   - Click "Raw" and save the file to `.bumpversion.toml` at your repository root
+
+#### Option C: Using Git (if you have both repositories locally)
+
+```bash
+cp path/to/s2dm/.migration/workflows/migrate.yml .github/workflows/migrate.yml
+cp path/to/s2dm/.migration/.bumpversion.toml .bumpversion.toml
+```
+
+#### Option D: Using curl
+
+```bash
+mkdir -p .github/workflows
+
+curl -o .github/workflows/migrate.yml https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/workflows/migrate.yml
+
+curl -o .bumpversion.toml https://raw.githubusercontent.com/COVESA/s2dm/main/.migration/.bumpversion.toml
+```
+
+### 2. Configure Repository Variables
+
+The workflow supports optional configuration through GitHub repository variables.
+
+Go to your repository → Settings → Secrets and variables → Actions → Variables tab, and add:
```

Comment:
What do they mean? Provide link or further information to the reader to understand the benefit of configuring this optional step. What implies to use and not to use them?